caddyhttp/fastcgi/fastcgi_test.go

```diff
@@ -51,6 +52,106 @@ func TestServeHTTP(t *testing.T) {
 	}
 }
 
+// listenerWithConnectionCounter adds a counter to keep track
+// of the number of accepted connections.
+type listenerWithConnectionCounter struct {
+	net.Listener
+	sync.RWMutex
```

Comment:
That's basically the same comment as https://github.com/mholt/caddy/pull/1134#discussion_r80407041, the read only lock is not needed in the first place. If both of you think that we should remove this, I'll obey. ;)